src/zarr/store/core.py

```diff
@@ -54,13 +56,11 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return f"StorePath({self.store.__class__.__name__}, {str(self)!r})"
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: object) -> bool:
         try:
-            if self.store == other.store and self.path == other.path:
-                return True
+            return bool(self.store == other.store and self.path == other.path)  # type: ignore[attr-defined]
```

Comment:
I had to add the `bool` cast to avoid this mypy error:
```src/zarr/store/core.py:61: error: Returning Any from function declared to return "bool"  [no-any-return]
src/zarr/store/core.py:61: note: Error code "no-any-return" not covered by "type: ignore" comment
```
I must admit I don't understand it. How could `__eq__` return `Any` instead of `bool`?

Reply:
@dstansby - any ideas here?

Reply:
Outdated: That code has disappeared.